client/promote_deployments.go

```diff
@@ -99,12 +93,59 @@ func PromoteDeployments(cli *Cli, commandOptions *CommandOptionsPromoteDeploymen
 			commandOptionsCreateDeployment.AppVersion[i] = actDeployment.AppsWithVersion[i].Version
 		}
 
-		deplyoment, err := CreateDeployment(cli, &commandOptionsCreateDeployment)
+		deployment, err := CreateDeployment(cli, &commandOptionsCreateDeployment)
 		if err != nil {
 			log.Println("Error Create Deployment: ", err)
 			return createdDeployments, err
 		}
-		createdDeployments = append(createdDeployments, *deplyoment)
+		createdDeployments = append(createdDeployments, *deployment)
+	}
+
+	//Wait on deplyoment success or failed
+	sleepTime := 60 //seconds, polling each x seconds
+	if commandOptions.Wait && commandOptions.MaxWaitTime > sleepTime {
+
+		commandOptionsGet := CommandOptionsGetDeployment{}
+		commandOptionsGet.Environment = []string{commandOptions.Environment}
+		commandOptionsGet.OnlyLatest = true
+		commandOptionsGet.TrackingID = -1
+		for _, actDeployment := range createdDeployments {
+			commandOptionsGet.AppServer = append(commandOptionsGet.AppServer, actDeployment.AppServerName)
+		}
+
+		//Timeout 10min = 600sec / 30sec = 20 counts
+		maxCounts := commandOptions.MaxWaitTime / sleepTime
+		for i := 0; i < maxCounts; i++ {
+			deployments, err := GetDeployment(cli, &commandOptionsGet)
+			if err != nil {
+				return nil, err
+			}
+
+			if len(deployments) == 0 {
+				log.Println("There was an error on creating the deplyoment, no deployment found from environment: ", commandOptions.FromEnvironment)
+				return nil, err
+			}
+
+			createdDeployments = deployments
+
+			// Check if all deployments are finished
+			allfinished := true
+			for _, actDeployment := range deployments {
```

Comment:
Move to separate methode for testing -> func allFinished(deployments Deployments) bool